src/main.js

```diff
@@ -138,3 +142,63 @@ const killAllRunningProcesses = () => {
     console.error('Got error when trying to kill children', err);
   }
 };
+
+const manageApplicationLocation = () => {
+  if (canApplicationBeMoved()) {
+    showMoveToApplicationsFolderDialog();
+  }
+};
+
+const canApplicationBeMoved = () => {
+  // The application can be moved if :
+  //  - The platform is MacOS
+  //  - The app is running in production
+  //  - The function 'isInApplicationsFolder' exists
+  //  - Guppy is not already in the Applications folder
+  const hasApplicationsFolder =
+    process.platform === 'darwin' &&
+    typeof app.isInApplicationsFolder === 'function';
+
+  const isProduction = process.env.NODE_ENV === 'production';
+
+  if (!hasApplicationsFolder || !isProduction || app.isInApplicationsFolder()) {
+    return false;
+  }
+
+  return true;
+};
+
+const showMoveToApplicationsFolderDialog = () => {
+  dialog.showMessageBox(
+    {
+      type: 'question',
+      buttons: ['Yes, move', 'Cancel'],
+      message: 'Move to applications folder?',
+      detail:
+        "I see that I'm not in the Applications folder. I can move myself there if you'd like!",
+      icon: path.join(__dirname, 'assets/icons/png/256x256.png'),
+      cancelId: 1,
+      defaultId: 0,
+      checkboxLabel: 'Do not show this message again',
+    },
+    (res, checkboxChecked) => {
+      // User wants the application to be moved
+      if (res === 0) {
+        try {
+          app.moveToApplicationsFolder();
+        } catch (err) {
+          dialog.showErrorBox(
+            'Error',
+            'Could not move Guppy to the Applications folder'
+          );
+          console.error('Could not move Guppy to the Applications folder', err);
+        }
+      }
+
+      // User doesn't want to see the message again
+      if (checkboxChecked === true) {
+        // TODO: Store choice in electron-store
+      }
```

Comment:
I was initially worried that this would be troublesome to add, but now that I've thought about it, I think it would work out fine.

We use [electron-store](https://github.com/sindresorhus/electron-store) for data persistence, and I was worried about possible conflicts, but all redux data is stored under a key, so as long as we pick another key, it _should_ work. It's still a bit unknown to me since we'd have to instantiate two different stores and it's not clear to me if doing so is problematic... but it should be fine!

So yeah, I'm more than happy to experiment with this myself after we merge this in, but if you were up for it, you could give it a try!

Essentially it would just be creating a new store, and `get`/`set`-ing a value like `leave-application-in-original-location` or something.